### PR TITLE
[ML][Data Frame] improve pivot nested field validations

### PR DESCRIPTION
**Pull request by benwtrent**
Generally, the aggregation validation contained here is good to catch overlapping data types
https://github.com/elastic/elasticsearch/blob/6c15c969e171a3801d9d6a464bae224870226153/x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/pivot/AggregationResultUtils.java#L111 

But, in more complicated scenarios where there are NO overlapping data types, it falls short. 

Let's say the user provides a `scripted_metric` aggregation that returns a `Map`. 
```
"super_metric": {
  "scripted_metric": {
    "init_script": "",
    "map_script": "",
    "combine_script": "",
    "reduce_script": "return ['value1': 1, 'value2':2]"
  }
}
```
So, in essence, `super_metric` is a nested field. If another agg is added with the name `super_metric.foo`, there is no return type overlap as in both cases, `super_metric` is an object that contains sub-fields. 

The validations done on `PUT` capture this scenario adequately, so I added them to `_preview`.

### CODE REVIEW

**Comment by benwtrent** [comment]
run elasticsearch-ci/1
run elasticsearch-ci/oss-distro-docs

File: x-pack/plugin/src/test/resources/rest-api-spec/test/data_frame/preview_transforms.yml [hunk 1]
```diff
@@ -203,6 +203,29 @@ setup:
...
           }
+  - do:
+      catch: /field \[super_metric\] cannot be both an object and a field/
+      data_frame.preview_data_frame_transform:
+        body: >
+          {
+            "source": { "index": "airline-data" },
+            "pivot": {
+              "group_by": {
+                "time": {"date_histogram": {"fixed_interval": "1h", "field": "time", "format": "yyyy-MM-DD HH"}}},
```
Comment (droberts195) [inline x-pack/plugin/src/test/resources/rest-api-spec/test/data_frame/preview_transforms.yml]: `DD` is creeping back into this file after it was corrected to `dd` in some places.  We can deal with this in another PR though, maybe by removing `format` altogether.  But please don't propagate `DD` anywhere else.

**Review by droberts195 — APPROVED** [review]
LGTM

File: x-pack/plugin/src/test/resources/rest-api-spec/test/data_frame/preview_transforms.yml (hunk 1 above)
Comment (benwtrent) [reply x-pack/plugin/src/test/resources/rest-api-spec/test/data_frame/preview_transforms.yml]: definitely, let me remove that stuff.

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Comment by benwtrent** [comment]
run elasticsearch-ci/2

**Pull request merged**